packages/on-chain/tasks/reality.ts

```diff
@@ -0,0 +1,523 @@
+    undefined,
+    types.string
+  )
+  .addParam(
+    "timeout",
+    "Timeout in seconds that should be required for the oracle",
+    48 * 3600,
+    types.int,
+    true
+  )
+  .addParam(
+    "cooldown",
+    "Cooldown in seconds that should be required after a oracle provided answer",
+    24 * 3600,
+    types.int,
+    true
+  )
+  .addParam(
+    "expiration",
+    "Time duration in seconds for which a positive answer is valid. After this time the answer is expired",
+    7 * 24 * 3600,
+    types.int,
+    true
+  )
+  .addParam("arbitrator", "Arbitrator address", undefined, types.string)
+  .addParam(
+    "bond",
+    "Minimum bond that is required for an answer to be accepted",
+    "0",
+    types.string,
+    true
+  )
+  .setAction(
+    async (taskArgs: RealityTaskArgs & { module: string }, hardhatRuntime) => {
+      await hardhatRuntime.run("verify", {
+        address: taskArgs.module,
+        constructorArgsParams: [
+          taskArgs.owner,
+          taskArgs.avatar,
+          taskArgs.target,
+          taskArgs.oracle,
+          `${taskArgs.timeout}`,
+          `${taskArgs.cooldown}`,
+          `${taskArgs.expiration}`,
+          `${taskArgs.bond}`,
+          taskArgs.template,
+          taskArgs.arbitrator,
+        ],
+      });
+    }
+  );
+
+task(
+  "reality:createDaoTemplate",
+  "Creates a question template on the oracle address"
+)
+  .addParam(
+    "oracle",
+    "Address of the oracle (e.g. Reality.eth)",
+    undefined,
+    types.string
+  )
+  .addParam(
+    "template",
+    "Template string for question (should include placeholders for proposal id and txs hash)",
+    JSON.stringify(defaultTemplate),
+    types.string,
+    true
+  )
+  .setAction(async (taskArgs, hardhatRuntime) => {
+    const [caller] = await hardhatRuntime.ethers.getSigners();
+    console.log("Using the account:", caller.address);
+    console.log("Creating template:", taskArgs.template);
+    const oracle = await hardhatRuntime.ethers.getContractAt(
+      "RealitioV3",
+      taskArgs.oracle
+    );
+
+    const receipt = await oracle
+      .createTemplate(JSON.stringify(defaultTemplate))
+      .then((tx: ContractTransaction) => tx.wait());
+
+    const id = receipt.logs[0].topics[1];
+    console.log("Template id:", id);
+    return id;
+  });
+
+task("reality:assemble", "Assembles a proposal for reality module")
+  .addParam("module", "Address of the reality module", undefined, types.string)
+  .addParam("shrine", "Address of the shrine", "undefined", types.string)
+  .addParam(
+    "ipfs",
+    "Ipfs hash that contains metadata for drop",
+    "undefined",
+    types.string
+  )
+  .addParam(
+    "token",
+    "Address of the token to be offered",
+    undefined,
+    types.string
+  )
+  .addParam(
+    "root",
+    "Merkle Root that includes all champions",
+    undefined,
+    types.string
+  )
+  .addParam("amount", "Amount of the token to be offered", undefined, types.int)
+  .setAction(
```

Comment:
I think this file gets quite large like this. It is not required on my side, seeing the time constraints, but I think it might make sense to add a `lib` directory with a file called `reality.ts` which has the different functions in it. Within that lib directory you could also abstract different things that you repeat in the tasks, like getting caller, the reality module instance etc.

Reply:
I looked through it and found `reatliy:assemble` a repetitive, so I refactoed. Can you give me an example of something else that you perceive as repetitive?

Reply:
I think what I had in mind was overkill for these tasks. Looks good to me the way it is